### ISSUE TITLE
Input shape mismatch occured for features in module EncDecDiarLabelModel

### ISSUE THREAD

**Issue by acampbellblack** [NVIDIA/NeMo#8276]
**Describe the bug**

Encountering this error when running the speaker diarization training tutorial notebook on my own data:
Input shape mismatch occured for features in module EncDecDiarLabelModel : 
Input shape expected = (batch, time) | 
Input shape found : torch.Size([2, 208000, 2])

**Steps/Code to reproduce bug**

I have run the data preparation scripts 'pathfiles_to_diarize_manifest.py' and 'create_msdd_train_dataset.py' on my own data.
My own data consists of a single .wav and .rttm file for each of 'train' and 'valid'.
The training rttm has 9 speakers.
The pairwise rttm files have been created successfully.
I am then loading the 'msdd_5scl_15_05_50Povl_256x3x32x2.yaml' config file and setting it up as follows:

name: MultiscaleDiarDecoder
sample_rate: 16000
num_workers: 20
batch_size: 5
model:
  diarizer:
    out_dir: null
    oracle_vad: true
    speaker_embeddings:
      model_path: titanet_large
      parameters:
        window_length_in_sec:
        - 1.5
        - 1.25
        - 1.0
        - 0.75
        - 0.5
        shift_length_in_sec:
        - 0.75
        - 0.625
        - 0.5
        - 0.375
        - 0.25
        multiscale_weights:
        - 1
        - 1
        - 1
        - 1
        - 1
        save_embeddings: true
  num_workers: 20
  max_num_of_spks: 2
  scale_n: 5
  soft_label_thres: 0.5
  emb_batch_size: 0
  train_ds:
    manifest_filepath: train/msdd_data.50step.json
    emb_dir: train
    sample_rate: ${sample_rate}
    num_spks: ${model.max_num_of_spks}
    soft_label_thres: ${model.soft_label_thres}
    labels: null
    batch_size: ${batch_size}
    emb_batch_size: ${model.emb_batch_size}
    shuffle: true
  validation_ds:
    manifest_filepath: valid/msdd_data.50step.json
    emb_dir: valid
    sample_rate: ${sample_rate}
    num_spks: ${model.max_num_of_spks}
    soft_label_thres: ${model.soft_label_thres}
    labels: null
    batch_size: 2
    emb_batch_size: ${model.emb_batch_size}
    shuffle: false
  test_ds:
    manifest_filepath: null
    emb_dir: null
    sample_rate: 16000
    num_spks: ${model.max_num_of_spks}
    soft_label_thres: ${model.soft_label_thres}
    labels: null
    batch_size: 2
    shuffle: false
    seq_eval_mode: false
  preprocessor:
    _target_: nemo.collections.asr.modules.AudioToMelSpectrogramPreprocessor
    normalize: per_feature
    window_size: 0.025
    sample_rate: ${sample_rate}
    window_stride: 0.01
    window: hann
    features: 80
    n_fft: 512
    frame_splicing: 1
    dither: 1.0e-05
  msdd_module:
    _target_: nemo.collections.asr.modules.msdd_diarizer.MSDD_module
    num_spks: ${model.max_num_of_spks}
    hidden_size: 256
    num_lstm_layers: 3
    dropout_rate: 0.5
    cnn_output_ch: 32
    conv_repeat: 2
    emb_dim: 192
    scale_n: 5
    weighting_scheme: conv_scale_weight
    context_vector_type: cos_sim
  loss:
    _target_: nemo.collections.asr.losses.bce_loss.BCELoss
    weight: null
  optim:
    name: adam
    lr: 0.001
    weight_decay: 0.001
    sched:
      name: CosineAnnealing
      min_lr: 1.0e-05
  speaker_model_cfg:
    model_defaults:
      filters: 1024
      repeat: 3
      dropout: 0.1
      separable: true
      se: true
      se_context_size: -1
      kernel_size_factor: 1.0
    preprocessor:
      _target_: nemo.collections.asr.modules.AudioToMelSpectrogramPreprocessor
      normalize: per_feature
      window_size: 0.025
      sample_rate: 16000
      window_stride: 0.01
      window: hann
      features: 80
      n_fft: 512
      frame_splicing: 1
      dither: 1.0e-05
    spec_augment:
      _target_: nemo.collections.asr.modules.SpectrogramAugmentation
      freq_masks: 3
      freq_width: 4
      time_masks: 5
      time_width: 0.03
    encoder:
      _target_: nemo.collections.asr.modules.ConvASREncoder
      feat_in: 80
      activation: relu
      conv_mask: true
      jasper:
      - filters: 1024
        repeat: 1
        kernel:
        - 3
        stride:
        - 1
        dilation:
        - 1
        dropout: 0.0
        residual: false
        separable: true
        se: true
        se_context_size: -1
      - filters: 1024
        repeat: 3
        kernel:
        - 7
        stride:
        - 1
        dilation:
        - 1
        dropout: 0.1
        residual: true
        separable: true
        se: true
        se_context_size: -1
      - filters: 1024
        repeat: 3
        kernel:
        - 11
        stride:
        - 1
        dilation:
        - 1
        dropout: 0.1
        residual: true
        separable: true
        se: true
        se_context_size: -1
      - filters: 1024
        repeat: 3
        kernel:
        - 15
        stride:
        - 1
        dilation:
        - 1
        dropout: 0.1
        residual: true
        separable: true
        se: true
        se_context_size: -1
      - filters: 3072
        repeat: 1
        kernel:
        - 1
        stride:
        - 1
        dilation:
        - 1
        dropout: 0.0
        residual: false
        separable: true
        se: true
        se_context_size: -1
    decoder:
      _target_: nemo.collections.asr.modules.SpeakerDecoder
      feat_in: 3072
      num_classes: 16681
      pool_mode: attention
      emb_sizes: 192
    loss:
      _target_: nemo.collections.asr.losses.angularloss.AngularSoftmaxLoss
      scale: 30
      margin: 0.2
    optim:
      name: sgd
      lr: 0.08
      weight_decay: 0.0002
      sched:
        name: CosineAnnealing
        warmup_ratio: 0.1
        min_lr: 0.0
      momentum: 0.9
    target: nemo.collections.asr.models.label_models.EncDecSpeakerLabelModel
    nemo_version: 1.23.0rc0
trainer:
  devices: 1
  accelerator: gpu
  max_epochs: 5
  max_steps: -1
  num_nodes: 1
  strategy: auto
  accumulate_grad_batches: 1
  deterministic: true
  enable_checkpointing: false
  logger: false
  log_every_n_steps: 1
  val_check_interval: 1.0
exp_manager:
  exp_dir: null
  name: ${name}
  create_tensorboard_logger: true
  create_checkpoint_callback: true
  create_wandb_logger: false
  checkpoint_callback_params:
    monitor: val_loss
    mode: min
    save_top_k: 30
    every_n_epochs: 1
  wandb_logger_kwargs:
    name: null
    project: null
emb_batch_size: 0

I am then attempting to run the trainer when encountering the above error.

**Environment overview (please complete the following information)**

 - Environment location: Bare-metal
 - Method of NeMo install: python -m pip install git+https://github.com/NVIDIA/NeMo.git@$BRANCH#egg=nemo_toolkit[all]

**Environment details**

- Windows 11 WSL2 Ubuntu 22.04.3
- torch=2.1.2+cu118
- Python 3.10.2

**Additional context**

GPU Model: Nvidia RTX 4080 Mobile

**Comment by acampbellblack** [comment]
Audio was stereo, my mistake
